File: app/src/test/java/net/tlalka/android/fiszki/model/dto/WordDtoTest.java
package net.tlalka.android.fiszki.model.dto;

import com.google.gson.Gson;
import net.tlalka.android.fiszki.model.dto.json.WordDto;
import net.tlalka.android.fiszki.model.types.LanguageType;
import org.junit.Before;
import org.junit.Test;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;

public class WordDtoTest {

    private Gson gson;

    @Before
    public void setup() {
        this.gson = new Gson();
    }

    @Test
    public void testParseJsonToWordDto() throws IOException {

        // given
        InputStream stream = getClass().getResourceAsStream("/samples/word.json");
        Reader reader = new InputStreamReader(stream, "UTF-8");

        // when
        WordDto dto = gson.fromJson(reader, WordDto.class);

        // then
        assertNotNull(dto);
        assertNotNull(dto.get(LanguageType.EN));
        assertNotNull(dto.get(LanguageType.PL));
        assertNotNull(dto.get(LanguageType.FR));
    }

    @Test
    public void testParseWordDtoToJson() throws IOException {

        // given
        WordDto dto = new WordDto();
        dto.put(LanguageType.EN, "dog");
        dto.put(LanguageType.PL, "pies");

        // when
        String result = gson.toJson(dto);

        // then
        assertNotNull(result);
        assertFalse(result.isEmpty());
    }
}

File: app/src/test/kotlin/net/tlalka/fiszki/model/dto/WordDtoTest.kt
package net.tlalka.fiszki.model.dto

import com.google.gson.Gson
import net.tlalka.fiszki.model.dto.json.WordDto
import net.tlalka.fiszki.model.types.LanguageType
import org.junit.Assert.assertFalse
import org.junit.Assert.assertNotNull
import org.junit.Test
import java.io.InputStreamReader

class WordDtoTest {

    @Test
    fun testParseJsonToWordDto() {

        // given
        val stream = javaClass.getResourceAsStream("/samples/word.json")
        val reader = InputStreamReader(stream, "UTF-8")

        // when
        val dto = Gson().fromJson(reader, WordDto::class.java)

        // then
        assertNotNull(dto)
        assertNotNull(dto[LanguageType.EN])
        assertNotNull(dto[LanguageType.PL])
        assertNotNull(dto[LanguageType.FR])
    }

    @Test
    fun testParseWordDtoToJson() {

        // given
        val dto = WordDto()
        dto[LanguageType.EN] = "dog"
        dto[LanguageType.PL] = "pies"

        // when
        val result = Gson().toJson(dto)

        // then
        assertNotNull(result)
        assertFalse(result.isEmpty())
    }
}
